heartbeat/ibmcloud-vpc-cr-vip.in

```diff
@@ -0,0 +1,248 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------
+import subprocess
+import ipaddress
+import os
+import sys
+import textwrap
+import time
+
+OCF_FUNCTIONS_DIR = os.environ.get(
+    "OCF_FUNCTIONS_DIR", "%s/lib/heartbeat" % os.environ.get("OCF_ROOT")
+)
+
+sys.path.append(OCF_FUNCTIONS_DIR)
+
+RETRY_COUNT = 1
+SLEEP_TIME  = 1
+
+try:
+    import ocf
+except ImportError:
+    sys.stderr.write("ImportError: ocf module import failed.")
+    sys.exit(5)
+
+try:
+    import ibm_cloud_fail_over
+except ImportError:
+     sys.stderr.write("ImportError: import ibm_cloud_fail_over module import failed.")
+     pass
+
+def os_ip(ip):
+    """Check if VSI own this IP address."""
+
+    command = ["ip", "a"]
+    response = subprocess.run(command, capture_output=True, text=True)
+    return ip in  response.stdout
+
+def ip_address_validator(ip):
+    """validate ip address string
+
+    Args:
+        ip (string): ip address
+
+    Returns:
+        bool: true if the strig is a valid ipv4 ip address
+    """
+    try:
+        ipaddress.ip_address(ip)
+        return True
+    except ValueError:
+        return False
+
+def validate_all_action(
+    vpc_url="",
+    ext_ip_1="",
+    ext_ip_2="",
+    api_key="",
+):
+    """validate all paramters
+
+    Args:
+        vpc_url (str, mandatory): vpc_url for teh region .
+        ext_ip_1 (str, mandatory): First Instance IP.
+        ext_ip_2 (str, mandatory): Secound Instance IP.
+        api_key (str, optional): IBM Cloud API acsess key. Defaults to "".
+
+    Returns:
+        _type_: _description_
+    """
+    if not ip_address_validator(ext_ip_1):
+      return ocf.OCF_ERR_CONFIGURED
+    if not ip_address_validator(ext_ip_2):
+      return ocf.OCF_ERR_CONFIGURED
+    ocf.logger.debug(f"validate_all_action: {vpc_url} {api_key}")
+    return ocf.OCF_SUCCESS
+
+def stop_action(
+    vpc_url="",
+    ext_ip_1="",
+    ext_ip_2="",
+    api_key="",
+):
+    """Stop VIP
+
+    Args:
+        vpc_url (str, mandatory): vpc_url for teh region .
+        ext_ip_1 (str, mandatory): First Instance IP.
+        ext_ip_2 (str, mandatory): Secound Instance IP.
+        api_key (str, optional): IBM Cloud API acsess key. Defaults to "".
+
+    Returns:
+        _type_: _description_
+    """
+    ocf.logger.info("stop_action:stoping")
+    res = monitor_action(vpc_url, ext_ip_1, ext_ip_2, api_key)
+    if res == ocf.OCF_NOT_RUNNING:
+        ocf.logger.info("Resource is already stopped")
```

Comment:
We need to run a delete action for the IPs here, or it wont be in a stopped state after stopping.